filament/src/ColorGrading.cpp

```diff
@@ -525,17 +545,41 @@ FColorGrading::FColorGrading(FEngine& engine, const Builder& builder) {
     //std::chrono::duration<float, std::milli> duration = std::chrono::steady_clock::now() - now;
     //slog.d << "LUT generation time: " << duration.count() << " ms" << io::endl;
 
-    TextureFormat textureFormat = selectLutTextureFormat(builder->quality);
+    TextureFormat textureFormat;
+    PixelDataFormat format;
+    PixelDataType type;
+    selectLutTextureParams(builder->quality, textureFormat, format, type);
 
     mLutHandle = driver.createTexture(SamplerType::SAMPLER_3D, 1, textureFormat, 0,
             config.lutDimension, config.lutDimension, config.lutDimension, TextureUsage::DEFAULT);
 
+    // convert input to UINT_2_10_10_10_REV if needed
+    if (type == PixelDataType::UINT_2_10_10_10_REV) {
+        uint32_t* const UTILS_RESTRICT dst = (uint32_t*)malloc(lutElementCount * sizeof(uint32_t));
```

Comment:
BTW we do we use mallocs in this file and now just new type[]?

Reply:
note sure. doesn't really matter.